pkg/registry/batch/cronjob/storage/storage.go

```diff
@@ -84,3 +101,53 @@ func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *me
 func (r *StatusREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
 	return r.store.Update(ctx, name, objInfo)
 }
+
+// InstantiateREST implements the REST endpoint for manually triggering a CronJob.
+type InstantiateREST struct {
+	cronJobStore *genericregistry.Store
+	jobStore     *genericregistry.Store
+}
+
+var _ = rest.NamedCreater(&InstantiateREST{})
+
+func (r *InstantiateREST) New() runtime.Object {
+	return &batch.CronJobManualInstantiation{}
+}
+
+func (r *InstantiateREST) Create(ctx genericapirequest.Context, name string, obj runtime.Object, includeUninitialized bool) (out runtime.Object, err error) {
+	sourceCronJobObj, err := r.cronJobStore.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	sourceCronJob, ok := sourceCronJobObj.(*batch.CronJob)
```

Comment:
I was hoping that `Create` will accept a `CronJobRequest` object and find the appropriate `CronJob` and return the actual `Job` as a result. See OpenShift's [build/instantiate](https://github.com/openshift/origin/blob/9f48098688b100e4d5f7b889818eab826eedb774/pkg/build/registry/buildconfiginstantiate/rest.go#L60-L75).

Reply:
As in, you think the subresource should operate on a completely different type of object? I felt that doing so was inelegant and would result in fighting against the API machinery, which I don't have enough knowledge about Kubernetes to know how to do. The NamedCreater interface made it straightforward to look up the source CronJob.

Reply:
The problem with this is that you need to pass the entire cronjob object just to instantiate it. Besides API server already knows the object. Besides if you're creating a Job it's better to create it from the CronJob that is on the server, not from the one you got, this might lead to differences you don't want to handle here.